### PR TITLE
Feature/sphinx

### PR DESCRIPTION
**Pull request by decaillet**

### CODE REVIEW

**Review by BenjaminDecaillet — APPROVED** [review]
Nice work! Sympa d'avoir mis mon nom en premier :p

File: doc/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+.. image:: img/functional-overview.png   
+   :alt: functional overview
+
+
+Evaluator
+---------
+The `Evaluator <apidoc/src.evaluator.html#src.evaluator.Evaluator>`_
+is an orchestrator that takes
+a `DataPreparator <apidoc/src.data_preparation.contract.html#src.data_preparation.contract.DataPreparator>`_,
```
Comment (BenjaminDecaillet) [inline doc/architecture.rst:21]: a `DataPreparator <apidoc/src.data_preparation.contract.html#src.data_preparation.contract.DataPreparator>`_, should be updtated to:
a `DataPreparator <apidoc/src.data_preparation.contract.html#src.preparator.contract.Preparator>`_,

File: doc/architecture.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Behind the scenes, `Service <apidoc/src.service.html#src.service.Service>`_ ensures the initialization of
+an  `Evaluator <apidoc/src.evaluator.html#src.evaluator.Evaluator>`_, via `factories <https://en.wikipedia.org/wiki/Factory_method_pattern>`_ 
+
+.. image:: img/service.png   
+   :alt: service
+
+NB: the above diagram shows the interactions of the following classes:
+
+- `ArgParser <apidoc/src.arg_parser.html#src.arg_parser.ArgParser>`_
+- `DataPreparatorFactory <apidoc/src.data_preparation.factory.html#src.data_preparation.factory.DataPreparatorFactory>`_
```
Comment (BenjaminDecaillet) [inline doc/architecture.rst:48]: In the same way :"- `DataPreparatorFactory <apidoc/src.data_preparation.factory.html#src.data_preparation.factory.DataPreparatorFactory>`_" should be changed to

- `DataPreparatorFactory <apidoc/src.data_preparation.factory.html#src.preaparator.factory.PreparatorFactory>`_

**Comment by BenjaminDecaillet** [comment]
Some changes should be done to match actual classes name

File: doc/architecture.rst (hunk 1 above)
Comment (BenjaminDecaillet) [reply doc/architecture.rst:21]: @decaillet  pourquoi tu le vois pas

**Comment by decaillet** [comment]
Un-merged... Will re-submit

**Pull request merged**